gradio/components.py

```diff
@@ -2761,6 +2761,138 @@ def style(self, *, full_width: Optional[bool] = None, **kwargs):
         return IOComponent.style(self, **kwargs)
 
 
+@document("change", "style")
+class UploadButton(Changeable, Uploadable, IOComponent, SimpleSerializable):
+    """
+    Used to create an upload button, when cicked allows a user to upload files that satisfy the specified file type or generic files (if file_type not set). The label is the text shown on the button, defaults to "Upload a File".
+    Preprocessing: passes the uploaded file as a {file-object} or {List[file-object]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
+    Postprocessing: expects function to return a {str} path to a file, or {List[str]} consisting of paths to files.
+    Examples-format: a {str} path to a local file that populates the component.
+    Demos:
+    """
+
+    def __init__(
+        self,
+        value: Optional[str | List[str] | Callable] = None,
+        *,
+        visible: bool = True,
+        elem_id: Optional[str] = None,
+        type: str = "file",
+        file_type: str = "file",
+        label: str = None,
+        **kwargs,
+    ):
+        """
+        Parameters:
+            value: Default file to display, given as str file path. If callable, the function will be called whenever the app loads to set the initial value of the component.
+            type: Type of value to be returned by component. "file" returns a temporary file object whose path can be retrieved by file_obj.name and original filename can be retrieved with file_obj.orig_name, "binary" returns an bytes object.
+            file_type: Type of file to be uploaded. "file" allows any file to be uploaded, "image" allows only image files to be uploaded, "audio" allows only audio files to be uploaded, "video" allows only video files to be uploaded, "text" allows only text files to be uploaded.
+            label: Text to display on the button. Defaults to "Upload a File".
```

Comment:
When I tested the `label` parameter, it seems like it wasn't having an effect:
<img width="722" alt="image" src="https://user-images.githubusercontent.com/1778297/202302590-96159a35-2a2b-4c8d-a140-3bb3857b7fce.png">